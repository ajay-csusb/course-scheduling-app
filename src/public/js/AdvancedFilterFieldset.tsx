import React from 'react';
import { Collapse, Label } from '@blueprintjs/core';
import { SelectListCourseAttr } from './SelectListCourseAttr';
import { SelectListSessionCode } from './SelectListSessionCode';
import { SelectListGeClassesAttributes } from './SelectListGeClassesAttributes';
import { IClassSearchFormProps } from './ClassSearchForm';

interface IAdvancedFilterFieldsetState {
  isOpen?: boolean;
}

export class AdvancedFilterFieldset extends React.Component<
    IClassSearchFormProps, IAdvancedFilterFieldsetState> {
  constructor(props: IClassSearchFormProps) {
    super(props);
    this.state = {
      isOpen: false,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  public render() {
    const courseAttr = this.getSelectListCourseAttrComponent();
    const classNumber = this.getClassNumberComponent();
    const sessionCode = this.getSelectListSessionCodeComponent();
    const geClassesAttribute = this.getSelectListGeClassesAttributesComponent();
    const arrow = (this.state.isOpen) ? <i className="fal fa-chevron-up"/> : <i className="fal fa-chevron-down"/>;
    return (
      <React.Fragment>
        <a id="additional-filters" onClick={this.handleClick}>
          {this.state.isOpen ? 'Hide' : ''} Additional Search Criteria {arrow}
        </a>
        <Collapse isOpen={this.state.isOpen} >
          <div className="row">
            <div className="col-6 col-md-4">
              <div className="form-item">{geClassesAttribute}</div>
            </div>
            <div className="col-6 col-md-4">
              <div className="form-item">{courseAttr}</div>
            </div>
            <div className="col-6 col-md-4">
              <div className="form-item">{classNumber}</div>
            </div>
            <div className="col-6 col-md-4">
              <div className="form-item">{sessionCode}</div>
            </div>
          </div>
        </Collapse>
      </React.Fragment>
    );
  }

  private handleClick = () => {
    this.setState({ isOpen: !this.state.isOpen });
  }

  private getSelectListCourseAttrComponent(): JSX.Element {
    return (
      <SelectListCourseAttr
        courseAttr={this.props.courseAttr}
        onChangeOfCourseAttr={this.props.onChangeOfCourseAttr}
      />
    );
  }

  private getClassNumberComponent(): JSX.Element {
    return (
      <React.Fragment>
        <Label htmlFor="class-number">Class Number</Label>
        <input
          id="class-number"
          className="class-number"
          type="text"
          placeholder="E.g. 60256"
          dir="auto"
          onChange={this.props.onChangeOfClassNo}
          value={this.props.classNo}
          onKeyDown={this.props.onKeyDown}
        />
      </React.Fragment>
    );
  }

  private getSelectListSessionCodeComponent(): JSX.Element {
    return (
      <SelectListSessionCode
        sessionCode={this.props.sessionCode}
        onChangeOfSessionCode={this.props.onChangeOfSessionCode}
      />
    );
  }

  private getSelectListGeClassesAttributesComponent(): JSX.Element {
    return (
      <SelectListGeClassesAttributes
        geClassesAttributes={this.props.geClassesAttributes}
        geClassesAttribute={this.props.geClassesAttribute}
        onChangeOfGeClassesAttribute={this.props.onChangeOfGeClassesAttribute}
      />
    );
  }

}
