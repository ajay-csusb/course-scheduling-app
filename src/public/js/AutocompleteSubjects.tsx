import { IItemRendererProps } from '@blueprintjs/select';
import React from 'react';
import { MenuItem } from '@blueprintjs/core';
import { ISubject } from './Subject';

export function filterSubjects(query: string, subject: ISubject): boolean {
  const parsedQuery = query.toLowerCase().trim();
  if (subject.name.toLowerCase().trim().indexOf(parsedQuery) >= 0) {
    return true;
  }
  if (subject.abbr.toLowerCase().trim().indexOf(parsedQuery) >= 0) {
    return true;
  }
  return false;
}

function renderSubjects(subject: ISubject, classRenderer: IItemRendererProps): JSX.Element {
  return (
    <MenuItem
      active={false}
      disabled={false}
      label={subject.abbr}
      key={subject.abbr}
      onClick={classRenderer.handleClick}
      text={subject.name}
      className="search-autocomplete-items"
    />
  );
}

export function renderSubject(subject: ISubject): string {
  return `${subject.name.trim()} (${subject.abbr.trim()})`;
}

export const autocompleteSubjectsProps = {
  itemRenderer: renderSubjects,
  itemPredicate: filterSubjects,
  inputValueRenderer: renderSubject,
};
