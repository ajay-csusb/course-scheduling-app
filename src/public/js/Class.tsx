import * as ClassSearchUtils from './ClassSearchUtils';
export interface IClass {
  quarter: string;
  subject: string;
  catalogNo: string;
  classSection: string;
  textbook: string;
  employeeId: string;
  profile?: string;
  classStartTime: string;
  classEndTime: string;
  courseOfferNo?: number;
  classNumber?: number;
  enrolledCapacity: number;
  enrolledTotal: number;
  waitlistCapacity: number;
  waitlistTotal: number;
  classMeetingNo: number;
  csuUnits: number;
  amount: number;
  courseId: string;
  sessionCode: string;
  description: string;
  ssrComponent: string;
  classType?: string;
  classStat?: string;
  campus: string;
  longDescription?: string;
  schedulePrint?: string;
  academicGroup?: string;
  academicOrg?: string;
  instructionMode: string;
  sbCourseZccm?: string;
  consent?: string;
  facilityId: string;
  buildingCode?: string;
  room?: string;
  mon: string;
  tues: string;
  wed: string;
  thurs: string;
  fri: string;
  sat: string;
  sun: string;
  instructorName: string;
  instructorEmployeeId?: string;
  instructorAltName?: string;
}

export interface IMeetingDate {
  all: boolean;
  mon: boolean;
  tue: boolean;
  wed: boolean;
  thu: boolean;
  fri: boolean;
  sat: boolean;
  sun: boolean;
}

export class Class {
  static classesUrl = 'http://webdx.csusb.edu/ClassSchedule/getCurrentCS';

  private classInfo: IClass;

  constructor(classData: IClass) {
    this.classInfo = classData;
  }

  static transformToClass(object: any): IClass {
    const result: IClass = {
      quarter : object.strm,
      subject : object.subject,
      catalogNo : object.catalog_NBR,
      classSection : object.class_SECTION,
      textbook : object.book,
      employeeId : object.emplid,
      profile : object.profile_PATH,
      classStartTime : object.class_START_TIME,
      classEndTime : object.class_END_TIME,
      courseOfferNo : object.crse_OFFER_NBR,
      classNumber : object.class_NBR,
      enrolledCapacity : object.enrl_CAP,
      enrolledTotal : object.enrl_TOT,
      waitlistCapacity : object.wait_CAP,
      waitlistTotal : object.wait_TOT,
      classMeetingNo : object.class_MTG_NBR,
      csuUnits : object.csu_APDB_CMP_UNITS,
      amount : object.flat_AMT,
      courseId : object.crse_ID,
      sessionCode : object.session_CODE,
      description : object.descr,
      ssrComponent : object.ssr_COMPONENT,
      classType : object.class_TYPE,
      classStat : object.class_STAT,
      campus : object.campus,
      longDescription : object.descrlong_VC,
      schedulePrint : object.schedule_PRINT,
      academicGroup : object.acad_GROUP,
      academicOrg : object.acad_ORG,
      instructionMode : object.instruction_MODE,
      sbCourseZccm : object.sb_CRSE_ZCCM,
      consent : object.consent,
      facilityId : object.facility_ID,
      buildingCode : object.bldg_CD,
      room : object.room,
      mon : object.mon,
      tues : object.tues,
      wed : object.wed,
      thurs : object.thurs,
      fri : object.fri,
      sat : object.sat,
      sun : object.sun,
      instructorName : object.name,
      instructorEmployeeId : object.emplid2,
      instructorAltName : object.name2,
    };
    return result;
  }

  static getAllClasses(onSuccess: (response: any) => void,
                       onFailure: (error: string) => void, url: string = this.classesUrl): void {
    const currentQuarterId = localStorage.getItem('currentQuarterId') ;
    const params = {
      currentPage: '1',
      maximunPerPage: '4000',
      conditions: ['strm = ' + currentQuarterId],
    };
    ClassSearchUtils.fetchWithArg(url, params, onSuccess, onFailure);
  }

  public getClassMeetingDates(): string {
    let days = '';
    let day = '';
    if (this.classInfo.mon === 'Y') {
      day += ' M';
    }
    if (this.classInfo.tues === 'Y') {
      day += ' Tu';
    }
    if (this.classInfo.wed === 'Y') {
      day += ' W';
    }
    if (this.classInfo.thurs === 'Y') {
      day += ' Th';
    }
    if (this.classInfo.fri === 'Y') {
      day += ' F';
    }
    if (this.classInfo.sat === 'Y') {
      day += ' Sa';
    }
    if (this.classInfo.sun === 'Y') {
      day += ' Su';
    }
    // Strip leading and trailing spaces, and replace spaces in between with -
    days = day.replace(/(^\s+)/g, '').replace(/(\s+)/g, '-');
    if (day === '') {
      days = 'N/A';
    }
    return days;
  }
  public getClassMeetingTimes(): string {
    let meetingTimes = 'N/A';
    if (!this.areStartTimeEndTimeEmpty()) {
      meetingTimes = `${this.classInfo.classStartTime} - ${this.classInfo.classEndTime}`.toLowerCase();
    }
    return meetingTimes;
  }

  private areStartTimeEndTimeEmpty(): boolean {
    return (this.classInfo.classStartTime.length === 0 && this.classInfo.classEndTime.length === 0);
  }
}
