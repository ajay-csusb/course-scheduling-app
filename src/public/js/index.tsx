import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ClassSearchContainer } from './ClassSearchContainer';
import { Utils } from './Utils';

Utils.loadExternalTools();
if (Utils.isProd()) {
  const ErrorBoundary = Utils.loadBugsnagComponent();
  ReactDOM.render(<ErrorBoundary><ClassSearchContainer /></ErrorBoundary>,
    document.querySelector('#class-search-form-wrapper'));
} else {
  ReactDOM.render(<ClassSearchContainer />, document.querySelector('#class-search-form-wrapper'));
}
