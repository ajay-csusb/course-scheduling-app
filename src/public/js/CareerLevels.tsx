import React from 'react';
import { ControlGroup, Checkbox, Alignment } from '@blueprintjs/core';
import { ICareerLevels } from './Class';

export interface ICareerLevelsProps {
  careerLevelsOptions: ICareerLevels;
  onChangeOfCareerLevelsOptions: (event: any) => void;
  onKeyDown: (event: React.KeyboardEvent) => void;
}

export class CareerLevels extends React.Component<ICareerLevelsProps, {}> {
  constructor(props: ICareerLevelsProps) {
    super(props);
    this.handleChangeOfCareerLevelsOptions = this.handleChangeOfCareerLevelsOptions.bind(this);
  }

  public render(): React.ReactNode {
    return (
      <ControlGroup fill={true} vertical={false} className="career-levels-filter-wrapper">
        <div className="label">Career Levels</div>
        <div id="career-levels-filter">
          <div className="form-item">
            <Checkbox
              id="ugrd"
              value="ugrd"
              className="career-levels-checkboxes"
              onChange={(event: React.FormEvent) => this.handleChangeOfCareerLevelsOptions(event)}
              checked={this.props.careerLevelsOptions.ugrd}
              onKeyDown={this.props.onKeyDown}
              inline={true}
              alignIndicator={Alignment.LEFT}
              label="Undergraduate"
            />
          </div>
          <div className="form-item">
            <Checkbox
              id="pbac"
              value="pbac"
              className="career-levels-checkboxes"
              onChange={(event: React.FormEvent) => this.handleChangeOfCareerLevelsOptions(event)}
              checked={this.props.careerLevelsOptions.pbac}
              onKeyDown={this.props.onKeyDown}
              inline={true}
              alignIndicator={Alignment.LEFT}
              label="Post baccalaureate"
            />
          </div>
          <div className="form-item">
            <Checkbox
              id="exed"
              value="exed"
              className="career-levels-checkboxes"
              onChange={(event: React.FormEvent) => this.handleChangeOfCareerLevelsOptions(event)}
              checked={this.props.careerLevelsOptions.exed}
              onKeyDown={this.props.onKeyDown}
              inline={true}
              alignIndicator={Alignment.LEFT}
              label="Extended education"
            />
          </div>
        </div>
      </ControlGroup>
    );
  }

  private handleChangeOfCareerLevelsOptions(event: React.FormEvent): void {
    this.props.onChangeOfCareerLevelsOptions(event);
  }
}
