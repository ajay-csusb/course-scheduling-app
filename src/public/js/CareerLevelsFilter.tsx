import { IClass, ICareerLevels } from './Class';

export function filter(classes: IClass[], careerLevelsOptions: ICareerLevels): IClass[] {
  const result: IClass[] = [];

  if (!atLeastOneOptionSelected(careerLevelsOptions)) {
    return classes;
  }

  classes.forEach(_class => {
    if (careerLevelsOptions.ugrd && _class.degreeType.toLowerCase() === 'ugrd') {
      result.push(_class);
    }
    if (careerLevelsOptions.pbac && _class.degreeType.toLowerCase() === 'pbac') {
      result.push(_class);
    }
    if (careerLevelsOptions.exed && _class.degreeType.toLowerCase() === 'exed') {
      result.push(_class);
    }
  });

  return result;
}

function atLeastOneOptionSelected(careerLevelsOptions: ICareerLevels) {
  return careerLevelsOptions.ugrd || careerLevelsOptions.pbac || careerLevelsOptions.exed;
}
