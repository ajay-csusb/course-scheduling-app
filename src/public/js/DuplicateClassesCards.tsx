import React from 'react';
import { IClass } from './Class';
import {
  getCourseHeaderMarkup,
  getCourseBodyMarkupForRepeatingClasses,
  getCourseFooterMarkup,
  getCourseInfoMarkup,
} from './ClassesCardsHelper';

interface IDuplicateClassesCardsProps {
  classes: IClass[];
  currentTerm: string;
}

export default function DuplicateClassesCards(duplicateClasses: IDuplicateClassesCardsProps): JSX.Element {
  const courseHeaderMarkup = getCourseHeaderMarkup(duplicateClasses.classes[0]);
  const courseInfoMarkup = getCourseInfoMarkup(duplicateClasses.classes[0], duplicateClasses.currentTerm);
  const courseBodyMarkup = getCourseBodyMarkupForRepeatingClasses(duplicateClasses.classes);
  const courseFooterMarkup = getCourseFooterMarkup(duplicateClasses.classes[0]);

  return (
    <div className="course result-item">
      <div className="item-header">
        {courseHeaderMarkup}
        {courseInfoMarkup}
      </div>
      <div className="item-body">
        {courseBodyMarkup}
        {courseFooterMarkup}
      </div>
    </div>
  );
}
