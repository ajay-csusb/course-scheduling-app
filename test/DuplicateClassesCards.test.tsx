import * as React from 'react';
import DuplicateClassesCards from '../src/public/js/DuplicateClassesCards';
import { classJson } from './ClassesJson';
import { mount } from 'enzyme';
import { TestUtils } from './TestUtils';
import { IClass } from '../src/public/js/Class';
import { Popover } from '@blueprintjs/core';

describe('when a class has multiple meeting times', () => {
    const copyClassJson: IClass = TestUtils.copyObject(classJson);
    copyClassJson.subject = 'BIOL';
    copyClassJson.catalogNo = '101';
    copyClassJson.classNumber = 100;
    copyClassJson.title = 'Introduction to Biology';
    copyClassJson.topic = 'Special topics in Biology';
    copyClassJson.courseAttr = 'Zero Cost Course Materials, Low Cost Course Materials';
    copyClassJson.fullSsrComponent = 'Seminar';
    copyClassJson.fee = '$100.00';
    copyClassJson.quarter = '0000';

    const bioClass1: IClass = TestUtils.copyObject(copyClassJson);
    bioClass1.classStartTime = '9:00 AM';
    bioClass1.classEndTime = '10:00 AM';
    bioClass1.mon = 'Y';
    bioClass1.campus = 'MAIN';
    const bioClass2: IClass = TestUtils.copyObject(copyClassJson);
    bioClass2.classStartTime = '11:00 AM';
    bioClass2.classEndTime = '12:00 PM';
    bioClass2.wed = 'Y';
    bioClass2.campus = 'PALM';
    const bioClass3: IClass = TestUtils.copyObject(copyClassJson);
    bioClass3.classStartTime = '1:00 PM';
    bioClass3.classEndTime = '3:00 PM';
    bioClass2.fri = 'Y';

  test('snapshot', () => {
    const classes: IClass[] = [bioClass1, bioClass2, bioClass3];
    const duplicateCardsComponent: JSX.Element = (
      <DuplicateClassesCards
        classes={classes}
        currentTerm={'000'}
      />
    );
    const duplicateCardsComponentWrapper = mount(duplicateCardsComponent);
    expect(duplicateCardsComponentWrapper).toMatchSnapshot();
  })

  it('should display each meeting time in a new row', () => {
    const classes: IClass[] = [bioClass1, bioClass2, bioClass3];
    const duplicateCardsComponent: JSX.Element = <DuplicateClassesCards classes={classes} currentTerm={'0000'} />;
    const duplicateCardsComponentWrapper = mount(duplicateCardsComponent);

    expect(duplicateCardsComponentWrapper.html()).toMatch(
      /BIOL 101 - Introduction to Biology: Special Topics In Biology/
    );
    expect(duplicateCardsComponentWrapper.html()).toMatch(/Section 01/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/Class No. 100/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/Fee/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/Seminar/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/Textbook/);
    expect(duplicateCardsComponentWrapper.find('.course-header')).toHaveLength(1);
    expect(duplicateCardsComponentWrapper.find('.course-details')).toHaveLength(1);
    expect(duplicateCardsComponentWrapper.find('.course-id')).toHaveLength(1);
    expect(duplicateCardsComponentWrapper.find('.course-name')).toHaveLength(1);
    expect(duplicateCardsComponentWrapper.find('.course-books')).toHaveLength(1);
    expect(duplicateCardsComponentWrapper.find('.course--icons')).toHaveLength(1);
    expect(duplicateCardsComponentWrapper.find('.course-desc')).toHaveLength(3);
    expect(duplicateCardsComponentWrapper.find(Popover)).toHaveLength(1);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/9:00 am - 10:00 am/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/11:00 am - 12:00 pm/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/1:00 pm - 3:00 pm/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/M/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/W/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/F/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/COE 101/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/San Bernardino/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/Palm Desert/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/Dyck, Harold/);
    expect(duplicateCardsComponentWrapper.html()).toMatch(/Classroom/)
    expect(duplicateCardsComponentWrapper.html()).toMatch(/2 \/ 30/);
  });
});
