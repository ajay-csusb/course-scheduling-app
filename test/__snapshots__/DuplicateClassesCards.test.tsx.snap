// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when a class has multiple meeting times snapshot 1`] = `
<DuplicateClassesCards
  classes={
    Array [
      Object {
        "amount": 0,
        "buildingCode": "COE",
        "campus": "MAIN",
        "catalogNo": "101",
        "classEndTime": "10:00 AM",
        "classMeetingNo": 1,
        "classNumber": 100,
        "classSection": "01",
        "classStartTime": "9:00 AM",
        "classStatus": "Active",
        "courseAttr": "Zero Cost Course Materials, Low Cost Course Materials",
        "courseAttrDescription": "",
        "courseId": "000001",
        "csuUnits": 4,
        "degreeType": "PBAC",
        "description": "ACCT GOVT & NONPROFIT ORG",
        "employeeId": "001343367",
        "enrolledCapacity": 30,
        "enrolledTotal": 28,
        "enrollmentStatus": "",
        "facilityId": "JB0113",
        "fee": "$100.00",
        "fri": "N",
        "fullSsrComponent": "Seminar",
        "geCourseAttr": "",
        "instructionMode": "P",
        "instructorName": "Dyck, Harold",
        "mon": "Y",
        "quarter": "0000",
        "room": "101",
        "sat": "N",
        "sessionCode": "1",
        "ssrComponent": "SEM",
        "subject": "BIOL",
        "sun": "N",
        "textbook": "",
        "thurs": "N",
        "title": "Introduction to Biology",
        "topic": "Special topics in Biology",
        "tues": "N",
        "waitlistCapacity": 100,
        "waitlistTotal": 1,
        "wed": "Y",
      },
      Object {
        "amount": 0,
        "buildingCode": "COE",
        "campus": "PALM",
        "catalogNo": "101",
        "classEndTime": "12:00 PM",
        "classMeetingNo": 1,
        "classNumber": 100,
        "classSection": "01",
        "classStartTime": "11:00 AM",
        "classStatus": "Active",
        "courseAttr": "Zero Cost Course Materials, Low Cost Course Materials",
        "courseAttrDescription": "",
        "courseId": "000001",
        "csuUnits": 4,
        "degreeType": "PBAC",
        "description": "ACCT GOVT & NONPROFIT ORG",
        "employeeId": "001343367",
        "enrolledCapacity": 30,
        "enrolledTotal": 28,
        "enrollmentStatus": "",
        "facilityId": "JB0113",
        "fee": "$100.00",
        "fri": "Y",
        "fullSsrComponent": "Seminar",
        "geCourseAttr": "",
        "instructionMode": "P",
        "instructorName": "Dyck, Harold",
        "mon": "Y",
        "quarter": "0000",
        "room": "101",
        "sat": "N",
        "sessionCode": "1",
        "ssrComponent": "SEM",
        "subject": "BIOL",
        "sun": "N",
        "textbook": "",
        "thurs": "N",
        "title": "Introduction to Biology",
        "topic": "Special topics in Biology",
        "tues": "N",
        "waitlistCapacity": 100,
        "waitlistTotal": 1,
        "wed": "Y",
      },
      Object {
        "amount": 0,
        "buildingCode": "COE",
        "campus": "MAIN",
        "catalogNo": "101",
        "classEndTime": "3:00 PM",
        "classMeetingNo": 1,
        "classNumber": 100,
        "classSection": "01",
        "classStartTime": "1:00 PM",
        "classStatus": "Active",
        "courseAttr": "Zero Cost Course Materials, Low Cost Course Materials",
        "courseAttrDescription": "",
        "courseId": "000001",
        "csuUnits": 4,
        "degreeType": "PBAC",
        "description": "ACCT GOVT & NONPROFIT ORG",
        "employeeId": "001343367",
        "enrolledCapacity": 30,
        "enrolledTotal": 28,
        "enrollmentStatus": "",
        "facilityId": "JB0113",
        "fee": "$100.00",
        "fri": "N",
        "fullSsrComponent": "Seminar",
        "geCourseAttr": "",
        "instructionMode": "P",
        "instructorName": "Dyck, Harold",
        "mon": "Y",
        "quarter": "0000",
        "room": "101",
        "sat": "N",
        "sessionCode": "1",
        "ssrComponent": "SEM",
        "subject": "BIOL",
        "sun": "N",
        "textbook": "",
        "thurs": "N",
        "title": "Introduction to Biology",
        "topic": "Special topics in Biology",
        "tues": "N",
        "waitlistCapacity": 100,
        "waitlistTotal": 1,
        "wed": "Y",
      },
    ]
  }
  currentTerm="000"
>
  <div
    className="course result-item"
  >
    <div
      className="item-header"
    >
      <div
        className="course-header"
      >
        <div
          className="course-name"
        >
          <strong>
            BIOL 101
             - Introduction to Biology: Special Topics In Biology
          </strong>
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            popoverClassName="bp3-popover-content-sizing"
            position="auto"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <Manager>
              <span
                className="bp3-popover-wrapper"
              >
                <Reference
                  innerRef={[Function]}
                >
                  <InnerReference
                    innerRef={[Function]}
                    setReferenceNode={[Function]}
                  >
                    <Blueprint3.ResizeSensor
                      onResize={[Function]}
                    >
                      <span
                        className="bp3-popover-target"
                        onClick={[Function]}
                      >
                        <div
                          className="bp3-tooltip-indicator"
                          key=".0"
                        >
                          <button
                            className="course-info-btn"
                          >
                            <i
                              arial-label="hidden"
                              className="fas fa-info-circle"
                            />
                            <span
                              className="sr-only"
                            >
                              BIOL 101
                            </span>
                          </button>
                        </div>
                      </span>
                    </Blueprint3.ResizeSensor>
                  </InnerReference>
                </Reference>
                <Blueprint3.Overlay
                  autoFocus={true}
                  backdropClassName="bp3-popover-backdrop"
                  backdropProps={Object {}}
                  canEscapeKeyClose={true}
                  canOutsideClickClose={true}
                  enforceFocus={true}
                  hasBackdrop={false}
                  isOpen={false}
                  lazy={true}
                  onClose={[Function]}
                  transitionDuration={300}
                  transitionName="bp3-popover"
                  usePortal={true}
                />
              </span>
            </Manager>
          </Blueprint3.Popover>
        </div>
        <div
          className="course-details"
        >
          <span>
            Section 
            01
             • 
          </span>
          <span>
            Seminar
             • 
          </span>
          <span
            className="course-id"
          >
            Class No. 
            100
          </span>
          <span>
             • Fees 
          </span>
           $
          $100.00
        </div>
      </div>
      <div
        className="course-info"
      >
        <div
          className="course-status course-status--closed"
        >
          <span />
        </div>
        <div
          className="course-availability"
        >
          Seats available:
           
          <span>
            2
             / 
            30
          </span>
        </div>
      </div>
    </div>
    <div
      className="item-body"
    >
      <div
        key="9:00 AM10:00 AM"
      >
        <ul
          className="course-desc"
        >
          <li>
            <span>
              Units 
            </span>
            4
          </li>
          <li>
            <span>
              Meeting Time 
            </span>
            9:00 am - 10:00 am
          </li>
          <li>
            <span>
              Meeting Days 
            </span>
            M-W
          </li>
          <li>
            <span>
              Room 
            </span>
            COE 101
          </li>
          <li>
            <span>
              Campus 
            </span>
            San Bernardino
          </li>
          <li>
            <span>
              Instructor
            </span>
            Dyck, Harold
          </li>
          <li>
            <span>
              Instruction Mode 
            </span>
            Classroom
          </li>
          <li>
             
            <span>
              Course Attribute 
            </span>
            Zero Cost Course Materials, Low Cost Course Materials
          </li>
        </ul>
      </div>
      <div
        key="11:00 AM12:00 PM"
      >
        <ul
          className="course-desc"
        >
          <li>
            <span>
              Units 
            </span>
            4
          </li>
          <li>
            <span>
              Meeting Time 
            </span>
            11:00 am - 12:00 pm
          </li>
          <li>
            <span>
              Meeting Days 
            </span>
            M-W-F
          </li>
          <li>
            <span>
              Room 
            </span>
            COE 101
          </li>
          <li>
            <span>
              Campus 
            </span>
            Palm Desert
          </li>
          <li>
            <span>
              Instructor
            </span>
            Dyck, Harold
          </li>
          <li>
            <span>
              Instruction Mode 
            </span>
            Classroom
          </li>
          <li>
             
            <span>
              Course Attribute 
            </span>
            Zero Cost Course Materials, Low Cost Course Materials
          </li>
        </ul>
      </div>
      <div
        key="1:00 PM3:00 PM"
      >
        <ul
          className="course-desc"
        >
          <li>
            <span>
              Units 
            </span>
            4
          </li>
          <li>
            <span>
              Meeting Time 
            </span>
            1:00 pm - 3:00 pm
          </li>
          <li>
            <span>
              Meeting Days 
            </span>
            M-W
          </li>
          <li>
            <span>
              Room 
            </span>
            COE 101
          </li>
          <li>
            <span>
              Campus 
            </span>
            San Bernardino
          </li>
          <li>
            <span>
              Instructor
            </span>
            Dyck, Harold
          </li>
          <li>
            <span>
              Instruction Mode 
            </span>
            Classroom
          </li>
          <li>
             
            <span>
              Course Attribute 
            </span>
            Zero Cost Course Materials, Low Cost Course Materials
          </li>
        </ul>
      </div>
      <div
        className="course-books"
      >
        <a
          className="btn btn-white btn-solid"
          href="https://www.bkstr.com/csusanbernardinostore/follett-discover-view/booklook?shopBy=discoverViewCourse&bookstoreId=1100&termId=0&divisionDisplayName=&departmentDisplayName=BIOL&courseDisplayName=101&sectionDisplayName=01"
          target="_blank"
        >
          <span
            className="sr-only"
          >
            BIOL
             
            101
          </span>
          Textbook
        </a>
        <div
          className="course--icons"
        >
          <img
            alt="Zero Cost Course Materials"
            src="https://www.csusb.edu/sites/default/files/zero-cost-book-icon-big.png"
            title="Zero Cost Course Materials"
          />
          <img
            alt="Low Cost Course Materials"
            src="https://www.csusb.edu/sites/default/files/low-cost-book-icon-big.png"
            title="Low Cost Course Materials"
          />
        </div>
      </div>
    </div>
  </div>
</DuplicateClassesCards>
`;
